apps/arweave/src/ar_info.erl

```diff
@@ -0,0 +1,94 @@
+%%%
+%%% @doc Gathers the data for the /info and /recent endpoints.
+%%%
+
+-module(ar_info).
+
+-export([get_info/0, get_recent/0]).
+
+-include_lib("arweave/include/ar.hrl").
+-include_lib("arweave/include/ar_chain_stats.hrl").
+
+get_info() ->
+	{Time, Current} =
+		timer:tc(fun() -> ar_node:get_current_block_hash() end),
+	{Time2, Height} =
+		timer:tc(fun() -> ar_node:get_height() end),
+	[{_, BlockCount}] = ets:lookup(ar_header_sync, synced_blocks),
+    #{
+        <<"network">> => list_to_binary(?NETWORK_NAME),
+        <<"version">> => ?CLIENT_VERSION,
+        <<"release">> => ?RELEASE_NUMBER,
+        <<"height">> =>
+            case Height of
+                not_joined -> -1;
+                H -> H
+            end,
+        <<"current">> =>
+            case is_atom(Current) of
+                true -> atom_to_binary(Current, utf8);
+                false -> ar_util:encode(Current)
+            end,
+        <<"blocks">> => BlockCount,
+        <<"peers">> => prometheus_gauge:value(arweave_peer_count),
+        <<"queue_length">> =>
+            element(
+                2,
+                erlang:process_info(whereis(ar_node_worker), message_queue_len)
+            ),
+        <<"node_state_latency">> => (Time + Time2) div 2
+    }.
+
+get_recent() ->
+    #{
+        %% #{
+        %%   "id": <indep_hash>,
+        %%   "received": <received_timestamp>"
+        %% }
+        <<"blocks">> => get_recent_blocks(ar_node:get_height()),
+        %% #{
+        %%   "id": <hash_of_block_ids>,
+        %%   "height": <height_of_first_orphaned_block>,
+        %%   "timestamp": <timestamp_of_when_fork_was_abandoned>
+        %%   "blocks": [<block_id>, <block_id>, ...]
+        %% }
+        <<"forks">> => get_recent_forks()
+    }.
+
+get_recent_blocks(CurrentHeight) ->
+    lists:foldl(
+        fun({H, _WeaveSize, _TXRoot}, Acc) ->
+            Acc ++ [#{
+                <<"id">> => ar_util:encode(H),
+                <<"received">> => get_block_timestamp(H, length(Acc))
+            }]
+        end,
+        [],
+        lists:sublist(ar_block_index:get_list(CurrentHeight), ?CHECKPOINT_DEPTH)
```

Comment:
ar_block_index:get_list is very slow as it fetches the entire block index - we may use e.g. [ar_node:get_block_anchors](https://github.com/ArweaveTeam/arweave/blob/master/apps/arweave/src/ar_node.erl#L183C1-L183C18) to quickly fetch the recent block hashes.

Reply:
Updated: https://github.com/ArweaveTeam/arweave/pull/581/commits/4764260fed3c794b8e5c17574edb43476ae51717